app/src/main/java/com/example/foodmanchu/IngredientsFragment.kt

```diff
@@ -2,29 +2,66 @@ package com.example.foodmanchu
 
 import android.os.Bundle
 import android.view.View
+import android.widget.Adapter
+import android.widget.Toast
 import androidx.fragment.app.Fragment
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.foodmanchu.databinding.IngredientsFragmentBinding
 
 class IngredientsFragment:Fragment(R.layout.ingredients_fragment) {
+
+    private var ingredientsAdapter: IngredientsAdapter ?= null
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         val binding = IngredientsFragmentBinding.bind(view)
 
-        var ingredientsAdapter = IngredientsAdapter()
-        val testing = ""
+        ingredientsAdapter = IngredientsAdapter()
+
         binding.apply {
             ingredientsRecyclerview.apply{
                 adapter = ingredientsAdapter
                 layoutManager = LinearLayoutManager(requireContext())
-                ingredientsAdapter.submitList(Repository.IngredientsList)
+                ingredientsAdapter!!.submitList(Repository.IngredientsList)
```

Comment:
code smell !!
```suggestion
                ingredientsAdapter!!.submitList(Repository.IngredientsList)
```

Reply:
added ? yeah better be safe

Reply:
I changed the adapter to late init var, better